tests/system/src/test/java/test/util/Util.java

```diff
@@ -466,4 +467,32 @@ public static boolean isFractionalScaleX(Node n) {
     private static boolean isFractional(double x) {
         return x != Math.rint(x);
     }
+
+    /**
+     * Returns the tolerance which should be used when comparing values,
+     * when {@link Region#isSnapToPixel()} returns true and the scale can
+     * be determined from the region's parent {@code Window}.
+     * When scale cannot be determined it is assumed to be 1.0.
+     * Otherwise, returns 0.0.
+     *
+     * @param r the region in question
+     * @return the tolerance value
+     */
+    public static double getTolerance(Region r) {
+        if (r.isSnapToPixel()) {
+            Scene scene = r.getScene();
+            if (scene != null) {
+                Window win = scene.getWindow();
+                if (win != null) {
+                    // x and y usually have the same scale, so we'll use x
+                    double scale = win.getRenderScaleX();
+                    // distance between pixels in the local (unscaled) coordinates is (1 / scale)
+                    return 1.0 / scale;
```

Comment:
this tolerance is to account for snapping, but it should be one half of the display pixel instead of the whole pixel.
With 0.5, these tests fail because we need JDK-8299753 to go first (and pass with the fix).

What we can do is to merge this change and change the tolerance as a part of JDK-8299753.

Reply:
My earlier comment about it being backwards was wrong. Since you are measuring and comparing user space pixels, those are indeed smaller for larger scale values. Ultimately, `0.5 / scale` would be a reasonable tolerance value.

> What we can do is to merge this change and change the tolerance as a part of JDK-8299753.

Sounds good to me.